aws_doc_sdk_examples_tools/ailly_prompt_generation/parse_json_files.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import json
+import logging
+from pathlib import Path
+from typing import List, Dict, Optional
+
+logging.basicConfig(level=logging.WARNING)
+logger = logging.getLogger(__name__)
+
+
+def extract_json_from_text(text: str) -> Optional[Dict]:
+    brace_stack: List[str] = []
+    json_start = None
+
+    for i, char in enumerate(text):
+        if char == "{":
+            if not brace_stack:
+                json_start = i
+            brace_stack.append(char)
+        elif char == "}":
+            if brace_stack:
+                brace_stack.pop()
+                if not brace_stack and json_start is not None:
+                    json_candidate = text[json_start : i + 1]
+                    try:
+                        return json.loads(json_candidate)
+                    except json.JSONDecodeError:
+                        logger.warning(
+                            "Found a brace-balanced block, but it isn't valid JSON."
+                        )
+                        # Continue searching
+    return None
+
+
+def process_files(file_paths: List[str]) -> List[Dict]:
```

Comment:
Doesn't this need to be `-> Dict[Path, Dict]` to line them back up later? Or some sort of ID to know which you just read back

Reply:
```json
  {
    "title": "Allows access to a specific Amazon Kinesis Data Firehose delivery stream",
    "title_abbrev": "Allows access to Firehose delivery stream",
    "description": "This example...",
    "path": ".ailly_prompts/snippet_files.iam-policies_latest.dg.prereqs-kinesis-data-firehose.xml.1.xml.md.ailly.md"
  },
```
What about that?

Reply:
Ok, didn't like that, went with your suggestion.
```
"snippet_files.iam-policies_latest.dg.configuring-dlq.xml.1.xml.md.ailly.md": {
    "title": "Allows Amazon SNS to send messages to an Amazon SQS dead-letter queue",
    "title_abbrev": "Allows SNS to send to SQS",
    "description": "This example..."
  },
  ```